clcache.py

```diff
@@ -126,6 +128,80 @@ def release(self):
         self._acquired = False
 
 
+class ObjectCacheLockFile:
+    """ Implements a lock file lock for the object cache which
+    can be used in 'with' statements.
+    Inspired by
+    https://github.com/harlowja/fasteners
+    https://github.com/benediktschmitt/py-filelock"""
+
+    def __init__(self, lockfileName, timeoutMs):
+        self._lockfileName = lockfileName
+        self._lockfile = None
+        self._timeoutMs = timeoutMs
+
+    def __enter__(self):
+        self.acquire()
+
+    def __exit__(self, typ, value, traceback):
+        self.release()
+
+    def __del__(self):
+        self.release()
+
+    def _acquire(self):
+        try:
+            lockfile = open(self._lockfileName, 'a')
+        except OSError:
+            lockfile = None
+        else:
+            try:
+                msvcrt.locking(lockfile.fileno(), msvcrt.LK_NBLCK, 1)
+            except (IOError, OSError):
+                lockfile.close()
+            else:
+                self._lockfile = lockfile
+        return None
+
+    def _release(self):
+        if self._lockfile is not None:
+            lockfile = self._lockfile
+            self._lockfile = None
+            msvcrt.locking(lockfile.fileno(), msvcrt.LK_UNLCK, 1)
+            lockfile.close()
+
+            #The following might fail because another instance already has locked the file.
+            #This is no problem because the existence of the file does not provide the
+            #locking but win32 api base file locking mechanism.
+            try:
+                os.remove(self._lockfileName)
+            except OSError:
+                pass
```

Comment:
Given that the existence of the file doesn't matter, I guess it could be argued that there's no point in removing the file - how about just dropping this code completely?